src/MapViewController.swift

```diff
@@ -417,11 +476,184 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
     tgViewController.markerRemove(marker)
     return
   }
+}
 
-  //MARK: - TGRecognizerDelegate
-
+extension MapViewController : TGMapViewDelegate, TGRecognizerDelegate {
+  
+  //MARK : TGMapViewDelegate
+  
+  open func mapView(_ mapView: TGMapViewController, didLoadSceneAsync scene: String) {
+    guard (loadDelegate != nil) && (self.loadDelegate?.responds(to: #selector(MapLoadDelegate.mapController(_:didLoadSceneAsync:))))! else {
```

Comment:
These all could be re-written more succinctly as such:
```
guard let unwrappedDelegate = loadDelegate,
  unwrappedDelegate.responds(to: #selector(MapLoadDelegate.mapController(_:didLoadSceneAsync:))) else { return }
```

This solves two issues: readability as I think it's much more readable, and secondly you don't ever have to force-unwrap anything (using `!` is generally frowned upon except in things like parsing code, and normally there's a better way). Also placing the return on the same line cuts down on line spam, but if you prefer it being on a separate line I'm open to reconsidering (compiled languages ftw).